### PR TITLE
Fix figure.colorbar() with axes keywords

### LINKED ISSUE #8493: Colorbar documentation: `anchor` not recognized as possible argument to plt.colorbar
The [figure api documentation of `colorbar`](http://matplotlib.org/api/figure_api.html#matplotlib.figure.Figure.colorbar) lists  `anchor` and `panchor` as
possible arguments to `figure.colorbar()`. However they do not seem to exist anymore. 

E.g. when running the following in matplotlib 2.0:
```
import matplotlib.pyplot as plt
import numpy as np; np.random.seed(1)

fig, ax = plt.subplots()
im = ax.imshow(np.random.rand(11,16))
ax.set_xlabel("x label")

fig.colorbar(im, ax=ax, anchor=(0.0, 0.5))
plt.show()
```

the following error is thrown:

```
Traceback (most recent call last):
  File "D:\⬜⬜⬜⬜\⬜⬜⬜\⬜⬜⬜\python\test\SO_colorbar.py", line 15, in <module>
    fig.colorbar(im, ax=ax, anchor=(0.0, 0.5))
  File "c:\⬜⬜⬜⬜\⬜⬜⬜⬜\python-2.7.10.amd64\lib\site-packages\matplotlib\figure.py", line 1602, in colorbar
    cb = cbar.colorbar_factory(cax, mappable, **kw)
  File "c:\⬜⬜⬜⬜\⬜⬜⬜⬜\python-2.7.10.amd64\lib\site-packages\matplotlib\colorbar.py", line 1347, in colorbar_factory
    cb = Colorbar(cax, mappable, **kwargs)
  File "c:\⬜⬜⬜⬜\⬜⬜⬜⬜\\python-2.7.10.amd64\lib\site-packages\matplotlib\colorbar.py", line 926, in __init__
    ColorbarBase.__init__(self, ax, **kw)
TypeError: __init__() got an unexpected keyword argument 'anchor'
```

I could imaging that this is simply because the documentation has not been updated?!

### PR DESCRIPTION
**Pull request by timhoffm**
Fixes #8493.

The problem was that `Figure.colorbar()` accepts `Axes` as well as `Colorbar` keywords. However they are all passed on to the `Colorbar` constructor. This has an explicit signature of supported keywords and complains on the `Axes` keywords 'anchor' and 'panchor'.

The fix is to filter out `Axes` keywords that are not allowed in `Colorbar`. Note: 'orientation' has to be passed to both.

### CODE REVIEW

**Comment by anntzer** [comment]
Congratulations on getting issue #10000 :-)

**Comment by timhoffm** [comment]
Thanks!

**Comment by jklymak** [comment]
Can we have a test that checks these kwargs are stripped so no one monkeys w/ this down the road?

**Comment by timhoffm** [comment]
@jklymak I've added a test. It just checks that the arguments are accepted and do not raise an exception.

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -1891,7 +1891,12 @@ def colorbar(self, mappable, cax=None, ax=None, use_gridspec=True, **kw):
             else:
                 cax, kw = cbar.make_axes(ax, **kw)
         cax._hold = True
-        cb = cbar.colorbar_factory(cax, mappable, **kw)
+
+        # need to remove kws that cannot be passed to Colorbar
+        NON_COLORBAR_KEYS = ['fraction', 'pad', 'shrink', 'aspect', 'anchor',
+                             'panchor']
```
Comment (dstansby) [inline lib/matplotlib/figure.py]: Does the remove of these kwargs mean the docstring was wrong/needs changing now?

Comment (timhoffm) [reply lib/matplotlib/figure.py]: I don't think so the notes section in the docs [says](http://matplotlib.org/api/_as_gen/matplotlib.figure.Figure.html?highlight=figure%20colorbar#matplotlib.figure.Figure.colorbar) says:

Additional keyword arguments are of two kinds:
- axes properties
- colorbar properties

We receive a mixture, which AFAICS is intended. But we have to remove the kwargs, that `Colorbar` does not understand.

**Pull request merged**